pkg/instrumentation/instrumentation.go

```diff
@@ -0,0 +1,96 @@
+package instrumentation
+
+import (
+	"context"
+	"fmt"
+
+	contribsdkconfig "go.opentelemetry.io/contrib/config"
+	sdklog "go.opentelemetry.io/otel/log"
+	sdkmetric "go.opentelemetry.io/otel/metric"
+	sdkresource "go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.26.0"
+	sdktrace "go.opentelemetry.io/otel/trace"
+	"go.uber.org/zap"
+)
+
+// Instrumentation holds the core components for application instrumentation.
+type Instrumentation struct {
+	LoggerProvider sdklog.LoggerProvider
+	Logger         *zap.Logger
+	MeterProvider  sdkmetric.MeterProvider
+	TracerProvider sdktrace.TracerProvider
+}
+
+// New creates a new Instrumentation instance with configured providers.
+// It sets up logging, tracing, and metrics based on the provided configuration.
+func New(ctx context.Context, build Build, cfg Config) (*Instrumentation, error) {
+	// Set default resource attributes if not provided
+	if cfg.Resource.Attributes == nil {
+		cfg.Resource.Attributes = map[string]any{
+			string(semconv.ServiceNameKey):    build.Name,
+			string(semconv.ServiceVersionKey): build.Version,
+		}
+	}
+
+	// Create a new resource with default detectors.
+	// The upstream contrib repository is not taking detectors into account.
+	// We are, therefore, using some sensible defaults here.
+	resource, err := sdkresource.New(
+		ctx,
+		sdkresource.WithContainer(),
+		sdkresource.WithFromEnv(),
+		sdkresource.WithProcess(),
+		sdkresource.WithTelemetrySDK(),
+		sdkresource.WithHost(),
```

Comment:
I suggest we remove process and telemetrysdk and host. We can always add them later if we need them because they are rarely used but consume a non-trivial amount of space. The container and env should be enough.

Reply:
Removed process and sdk. I guess host would help in debugging?